corehq/apps/geospatial/tests/test_index_geolocation_case_properties.py

```diff
@@ -0,0 +1,76 @@
+from django.test import TestCase
+
+from casexml.apps.case.mock import CaseFactory
+
+from corehq.apps.es import CaseSearchES
+from corehq.apps.es.case_search import case_search_adapter
+from corehq.apps.es.tests.utils import es_test
+from corehq.apps.geospatial.management.commands.index_geolocation_case_properties import (
+    _case_query,
+    index_case_docs,
+)
+from corehq.apps.geospatial.models import GeoConfig
+
+
+@es_test(requires=[case_search_adapter], setup_class=True)
+class TestGetFormCases(TestCase):
+    domain = 'foobar'
+    primary_case_type = 'primary'
+    secondary_case_type = 'secondary'
+    gps_prop_name = 'gps_coordinates'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        factory = CaseFactory(cls.domain)
+        case1 = factory.create_case(
+            case_name='foo',
+            case_type=cls.primary_case_type,
+            update={cls.gps_prop_name: '1.1 2.2'},
+        )
+        case2 = factory.create_case(
+            case_name='bar',
+            case_type=cls.primary_case_type,
+            update={cls.gps_prop_name: '3.3 4.4'},
+        )
+        case3 = factory.create_case(
+            case_name='other',
+            case_type=cls.secondary_case_type,
+            update={cls.gps_prop_name: '5.5 6.6'},
+        )
+        case_search_adapter.bulk_index([case1, case2, case3], refresh=True)
+        cls.geo_config = GeoConfig.objects.create(
+            domain=cls.domain,
+            case_location_property_name=cls.gps_prop_name
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.geo_config.delete()
```

Comment:
nit: consider skipping this as SQL objects are automatically as part of teardown in TestCase.

Reply:
Addressed in [807e5c8](https://github.com/dimagi/commcare-hq/pull/35090/commits/807e5c8c16605a3f625f5c82cc998d0588e55b1d).

Reply:
I think it would be good to have the teardown as geo_config is cached in redis and `geo_config.delete` clears that. Maybe having it in `setUpClass` like -  `self.addClassCleanup(cls.geo_config, delete)` 

Although it won't matter in test runs in CI but for local runs I think it would fail the second time 🤔

Reply:
Testing this out locally @AmitPhulera is correct. This does fail on running the test locally for a second time. I have addressed this by adding the necessary cleanup call in the setup class ([bc3f05d](https://github.com/dimagi/commcare-hq/pull/35090/commits/bc3f05d20b1228930f74e4da98913b96e64c4412)).